pkg/processor/runtime/rpc/abstract.go

```diff
@@ -685,3 +685,30 @@ func (r *AbstractRuntime) waitForProcessTermination(timeout time.Duration) {
 		}
 	}
 }
+
+func (r *AbstractRuntime) waitForDrainingFinish(drainDoneControlMessageChan chan *controlcommunication.ControlMessage, timeout time.Duration) {
+	r.Logger.DebugWith("Waiting for draining to be finished",
+		"wid", r.Context.WorkerID,
+		"process", r.wrapperProcess,
+		"timeout", timeout.String())
+
+	for {
+		select {
+		case <-r.stopChan:
+			r.Logger.DebugWith("Process terminated",
+				"wid", r.Context.WorkerID,
+				"process", r.wrapperProcess)
+			return
```

Comment:
Now there's no real reason we should be terminated in case of draining.
If we do it means something real fishy happened, but I don't think we should wait on it.